### PR TITLE
Added instructions to fetch face detection dataset and licenses to newly added files

### PR DESCRIPTION
**Pull request by krantikiran68**
Added fetchFDDatasets.py that uses the requests python library to fetch the datasets from krantikiran68/EdgeML github repository. 
Added license statemet in architecture.md.

### CODE REVIEW

**Comment by krantikiran68** [comment]
Added the instructions to fetch the face-2 dataset by allowing dataset as a class attribute in fetchFDDataset.py

File: tools/SeeDot/seedot/compiler/input/rnnpool-face-2.sd [hunk 1]
```diff
@@ -0,0 +1,355 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT license.
+
+// RNNPool SeeDot code for face-4 dataset
```
Comment (ShikharJ) [inline tools/SeeDot/seedot/compiler/input/rnnpool-face-2.sd]: Face-2 here, no?

File: tools/SeeDot/faceDetection.md [hunk 2]
```diff
@@ -0,0 +1,114 @@
+# Quantizing Face Detection using SeeDot
+
+Face Detection using SeeDot can be performed on the `face-2` and `face-4` datasets, which are processed subsets of the [SCUT Heads](https://github.com/HCIILAB/SCUT-HEAD-Dataset-Release) dataset.
```
Comment (ShikharJ) [inline tools/SeeDot/faceDetection.md]: `SCUT Head` here instead of `SCUT Heads`.

File: tools/SeeDot/faceDetection.md [hunk 3]
```diff
@@ -0,0 +1,114 @@
...
+
+Face Detection is a regression problem that involves an image input with (or without) faces and bounding boxes around the faces as output. 
+Face Detection is supported for x86 and M3 target devices.
```
Comment (ShikharJ) [inline tools/SeeDot/faceDetection.md]: Replace "M3" with "ARM Cortex-M3".

File: tools/SeeDot/faceDetection.md [hunk 4]
```diff
@@ -0,0 +1,114 @@
+# Quantizing Face Detection using SeeDot
```
Comment (ShikharJ) [inline tools/SeeDot/faceDetection.md]: Replace "Quantizing" with "Quantized" here please.

File: tools/SeeDot/faceDetection.md [hunk 5]
```diff
@@ -0,0 +1,114 @@
...
+
+Face Detection using SeeDot can be performed on the `face-2` and `face-4` datasets, which are processed subsets of the [SCUT Heads](https://github.com/HCIILAB/SCUT-HEAD-Dataset-Release) dataset.
+
+Face Detection is a regression problem that involves an image input with (or without) faces and bounding boxes around the faces as output. 
+Face Detection is supported for x86 and M3 target devices.  
+
+### Obtaining the datasets
```
Comment (ShikharJ) [inline tools/SeeDot/faceDetection.md]: Let's keep the headings capitalized with the first letter of the words.

File: tools/SeeDot/faceDetection.md [hunk 6]
```diff
@@ -0,0 +1,114 @@
...
+To obtain the model weights and datasets for the face detection problem, run `fetchFDDataset.py` in `seedot/compiler/input/` folder.
+Or simply run the commands from SeeDot's home directory (`EdgeML/tools/SeeDot/`):
```
Comment (ShikharJ) [inline tools/SeeDot/faceDetection.md]: "Run the following commands" here please.

File: tools/SeeDot/faceDetection.md [hunk 7]
```diff
@@ -0,0 +1,114 @@
...
+```
+    cd seedot/compiler/input/
+    python fetchFDDataset.py
+```
+
+This python script will place the model weights in `model/rnnpool/face-2/` for the `face-2` dataset and in `model/rnnpool/face-2/` for the `face-4` dataset.
```
Comment (ShikharJ) [inline tools/SeeDot/faceDetection.md]: `model/rnnpool//face-4` for the `face-4` here please.

File: tools/SeeDot/faceDetection.md [hunk 8]
```diff
@@ -0,0 +1,114 @@
...
+The datasets are placed in `datasets/rnnpool/face-2/` for the `face-2` dataset and in `datasets/rnnpool/face-2/` for the `face-4` dataset.
```
Comment (ShikharJ) [inline tools/SeeDot/faceDetection.md]: `datasets/rnnpool/face-4` for the `face-4` here please.

File: tools/SeeDot/faceDetection.md [hunk 9]
```diff
@@ -0,0 +1,114 @@
...
+
+### Run Face Detection for x86
+To run face detection using the SeeDot quantizer for x86 devices, run the command:
```
Comment (ShikharJ) [inline tools/SeeDot/faceDetection.md]: Replace "for" with "on" here please.

File: tools/SeeDot/faceDetection.md [hunk 10]
```diff
@@ -0,0 +1,114 @@
...
+```
+    python SeeDot-dev.py -a rnnpool -e fixed -m disagree -d face-4 -dt testing -t m3 -n 18000 
+```
+for the `face-4` dataset.
+
+Note: The target device specified using `-t` argument has the value `m3` in this case. See above for detailed argument description.
+
+The output will be stored to the `m3dump/` directory by default. Use the `-o <destination folder>` flag to store the output to an already existing location. 
+
+## Run SeeDot's output quantized code
```
Comment (ShikharJ) [inline tools/SeeDot/faceDetection.md]: Same for this heading please.

File: tools/SeeDot/seedot/compiler/input/fetchFDDataset.py [hunk 11]
```diff
@@ -0,0 +1,191 @@
...
+
+
+import os
+import requests
+import subprocess
+from tqdm import tqdm
+import numpy as np
+
+'''
+The python file to obtain the dataset for face Detection.
```
Comment (ShikharJ) [inline tools/SeeDot/seedot/compiler/input/fetchFDDataset.py]: Capitalize the first letter of "face" here please.

File: tools/SeeDot/seedot/compiler/input/fetchFDDataset.py [hunk 12]
```diff
@@ -0,0 +1,191 @@
...
+    def copySeeDotfile(self):
+        os.popen("cp rnnpool-%s.sd %s/input.sd"%(self.dataset, self.modelDir))
+    
+    def checkCorrectDir(self):
+        errormessage = \
+            "Run this python script from ${EdgeML Repo Folder}/tools/SeeDot/seedot/compiler/input/ directory..."
+        
+        if not ("rnnpool-%s.sd"%(self.dataset) in os.listdir(os.getcwd())):
+            assert False, errormessage
+
```
Comment (ShikharJ) [inline tools/SeeDot/seedot/compiler/input/fetchFDDataset.py:175]: Please add a newline here.

File: tools/SeeDot/seedot/compiler/input/fetchFDDataset.py [hunk 13]
```diff
@@ -0,0 +1,191 @@
...
+def run(obj):
+    obj.checkCorrectDir()
+
+    obj.createDirectories()
+    obj.fetchModelFiles()
+    obj.fetchDatasetFiles()
+    obj.copySeeDotfile()
+
```
Comment (ShikharJ) [inline tools/SeeDot/seedot/compiler/input/fetchFDDataset.py:184]: And here.

File: tools/SeeDot/seedot/compiler/input/fetchFDDataset.py [hunk 14]
```diff
@@ -0,0 +1,191 @@
...
+if __name__ == '__main__':
+    obj = FetchFaceDetectionDataset("face-4")
+    run(obj)
+
+    obj = FetchFaceDetectionDataset("face-2")
+    run(obj)
+
+
```
Comment (ShikharJ) [inline tools/SeeDot/seedot/compiler/input/fetchFDDataset.py]: Please remove this newline.

**Review by ShikharJ — COMMENTED** [review]
Minor comments.

File: tools/SeeDot/faceDetection.md (hunk 4 above)
Comment (krantikiran68) [reply tools/SeeDot/faceDetection.md]: Since the document is discussing the steps for quantizing won't 'Quantizing' sound better?

Comment (ShikharJ) [reply tools/SeeDot/faceDetection.md]: Nope. Quantization is a process. You quantize neural networks (an object), how would you quantize face detection (a problem/concept)?

File: tools/SeeDot/faceDetection.md (hunk 10 above)
Comment (krantikiran68) [reply tools/SeeDot/faceDetection.md]: I didn't understand

Comment (ShikharJ) [reply tools/SeeDot/faceDetection.md]: Run Seedot's Output Quantized Code

File: tools/SeeDot/faceDetection.md (hunk 5 above)
Comment (ShikharJ) [reply tools/SeeDot/faceDetection.md]: This is still unchanged.

File: tools/SeeDot/faceDetection/layers/modules/__init__.py [hunk 15]
```diff
@@ -0,0 +1,8 @@
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT license.
+
+from .l2norm import L2Norm
+from .multibox_loss import MultiBoxLoss
+
+__all__ = ['L2Norm', 'MultiBoxLoss']
+
```
Comment (ShikharJ) [inline tools/SeeDot/faceDetection/layers/modules/__init__.py]: Please leave only a single newline at the bottom.

File: tools/SeeDot/faceDetection.md (hunk 5 above)
Comment (krantikiran68) [reply tools/SeeDot/faceDetection.md]: My bad

**Review by ShikharJ — COMMENTED** [review]
LGTM

**Comment by ShikharJ** [comment]
Thanks @krantikiran68 for the contribution.

**Pull request merged**